contracts/scripts/testLocalExecuteBoldUpgrade.ts

```diff
@@ -1,88 +1,173 @@
-import { AbiCoder } from 'ethers/lib/utils'
+import { BOLDUpgradeAction, RollupMigratedEvent } from '../build/types/src/rollup/BOLDUpgradeAction.sol/BOLDUpgradeAction'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { getAddress } from 'ethers/lib/utils'
 
 dotenv.config()
 
-async function main() {
-  const l1RpcVal = process.env.L1_RPC_URL
-  if (!l1RpcVal) {
-    throw new Error('L1_RPC_URL env variable not set')
-  }
-  const l1Rpc = new ethers.providers.JsonRpcProvider(l1RpcVal)
-
-  const l1PrivKey = process.env.L1_PRIV_KEY
-  if (!l1PrivKey) {
-    throw new Error('L1_PRIV_KEY env variable not set')
-  }
-  const wallet = new Wallet(l1PrivKey, l1Rpc)
-
-  const deployedContractsLocation = process.env.DEPLOYED_CONTRACTS_LOCATION
-  if (!deployedContractsLocation) {
-    throw new Error('DEPLOYED_CONTRACTS_LOCATION env variable not set')
-  }
-  const configLocation = process.env.CONFIG_LOCATION
-  if (!configLocation) {
-    throw new Error('CONFIG_LOCATION env variable not set')
-  }
-  const config = await getConfig(configLocation, l1Rpc)
+async function perform(l1Rpc: JsonRpcProvider, config: Config, deployedContracts: DeployedContracts) {
+  await l1Rpc.send(
+    "hardhat_impersonateAccount",
+    ["0xE6841D92B0C345144506576eC13ECf5103aC7f49".toLowerCase()],
+  )
 
-  const deployedContracts = getJsonFile(
-    deployedContractsLocation
-  ) as DeployedContracts
-  if (!deployedContracts.boldAction) {
-    throw new Error('No boldAction contract deployed')
-  }
+  await l1Rpc.send(
+    "hardhat_setBalance",
+    ["0xE6841D92B0C345144506576eC13ECf5103aC7f49", '0x1000000000000000'],
+  )
 
+  const timelockImposter = l1Rpc.getSigner('0xE6841D92B0C345144506576eC13ECf5103aC7f49'.toLowerCase())
+  
   const upExec = new Contract(
     config.contracts.upgradeExecutor,
     UpgradeExecutorAbi,
-    wallet
+    timelockImposter
   )
   const boldAction = BOLDUpgradeAction__factory.connect(
     deployedContracts.boldAction,
-    wallet
+    timelockImposter
   )
 
-  // set config.validators in old rollup
-  const setValidatorCalldata = RollupAdminLogic__factory.createInterface().encodeFunctionData('setValidator', [config.validators, Array(config.validators.length).fill(true)])
-  const executeCallCalldata = ethers.utils.concat(['0xbca8c7b5', new AbiCoder().encode(['address', 'bytes'], [config.contracts.rollup, setValidatorCalldata])])
-  await (await wallet.sendTransaction({
-    to: upExec.address,
-    data: executeCallCalldata,
-  })).wait()
-
   // what validators did we have in the old rollup?
   const boldActionPerformData = boldAction.interface.encodeFunctionData(
     'perform',
     [config.validators]
   )
 
-  const receipt = (await (
+  return (await (
     await upExec.execute(deployedContracts.boldAction, boldActionPerformData)
   ).wait()) as ContractReceipt
+}
+
+async function verifyPostUpgrade(l1Rpc: JsonRpcProvider, config: Config, deployedContracts: DeployedContracts, receipt: ContractReceipt) {
```

Comment:
Can we also verify that the stakers have been moved over correctly?

Reply:
sounds good, i'm working on another branch that extends this test